apps/server/internal/app/runtime_instances.go

```
@@ -0,0 +1,260 @@
+func NewRuntimeInstanceService(runtimeStore RuntimeInstanceStore, workspaces IssueWorkspaceEnsurer, implementations map[string]runtimepkg.Implementation) (*RuntimeInstanceService, error) {
+	if runtimeStore == nil || workspaces == nil {
+		return nil, fmt.Errorf("runtime instance service dependencies are required")
+	}
+	if len(implementations) == 0 {
+		return nil, fmt.Errorf("at least one Runtime implementation is required")
+	}
+	copyImplementations := make(map[string]runtimepkg.Implementation, len(implementations))
+	for kind, implementation := range implementations {
+		kind = strings.TrimSpace(kind)
+		if kind == "" || implementation == nil {
+			return nil, fmt.Errorf("Runtime implementation kind and value are required")
```

golangci-lint (2.13.2)

[error] ST1005: error strings should not be capitalized

(staticcheck)